File: task-03/HelloWorld.go
//GO program to print all the prime numbers upto n

package main
import ("fmt")

func main() {
	var n int

	fmt.Println("Enter a number: ")
	fmt.Scan(&n)

	for i := 2; i <= n; i++ {
		if isPrime(i) {
			fmt.Println(i)
		}
	}
}

func isPrime(num int) bool {
	if num < 2 {
		return false
	}

	for i := 2; i*i <= num; i++ {
		if num%i == 0 {
			return false
		}
	}

	return true
}

File: task-03/HelloWorld.rs
//Rust program to print all the prime numbers upto n 


fn is_prime(num: u32) -> bool {
    if num <= 1 {
        return false;
    }
    if num <= 3 {
        return true;
    }
    if num % 2 == 0 || num % 3 == 0 {
        return false;
    }
    let mut i = 5;
    while i * i <= num {
        if num % i == 0 || num % (i + 2) == 0 {
            return false;
        }
        i += 6;
    }
    true
}

fn print_primes_upto(n: u32) {
    for i in 2..=n {
        if is_prime(i) {
            println!("{}", i);
        }
    }
}

fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: u32 = match input.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Invalid input. Please enter a valid number.");
            return;
        }
    };

    print_primes_upto(n);
}


   

